app/models/question.rb

```diff
@@ -0,0 +1,50 @@
+class Question < ApplicationRecord
+  include ::TokenHandler
+
+  belongs_to :user
+
+  before_create -> { generate_token(:published_token) }
+
+  before_save :ensure_published_question_cannot_be_drafted
+
+  with_options if: :published? do
+    validates :title, presence: true, uniqueness: true
+    validates :content, presence: true, length: { minimum: 15 }
+    validates :topic_list, presence: true
+  end
+
+  has_one_attached :pdf_attachment
+  has_rich_text :content
+  acts_as_taggable_on :topics
+
+  private def ensure_published_question_cannot_be_drafted
+    if changes[:published] == [true, false]
+      errors.add(:base, "Published question cannot be Drafted")
+      throw :abort
+    end
+  end
+
+  def to_param
+    published_token
+  end
+  
+  def published?
+    return !(self.published_at.nil?)
+  end
+
+  def publish_question(published)
+    check_publish(published)
```

Comment:
why are we passing published as an argument here?
check in controller if this is true and call publish